Tests/CombineWampTests/IntegrationTests/SubscriberTests.swift

```diff
@@ -0,0 +1,115 @@
+import Combine
+@testable import CombineWamp
+import Foundation
+import XCTest
+
+// https://wamp-proto.org/_static/gen/wamp_latest.html#subscribing-and-unsubscribing
+// Apart from the Router, in order to run this test you also must run the publishing Docker
+// https://crossbar.io/docs/Getting-Started/#publishing-client
+// All of this will eventually be automated.
+final class SubscriberTests: IntegrationTestBase {
+    var connected: XCTestExpectation!
+    var session: WampSession!
+    let realm = URI("realm1")!
+    var connection: AnyCancellable?
+
+    override func setUp() {
+        super.setUp()
+
+        connected = expectation(description: "Connected")
+        session = WampSession(transport: transport(), serialization: serialization, realm: realm, roles: .allClientRoles)
+        connection = session.connect()
+            .sink(
+                receiveCompletion: { completion in
+                    switch completion {
+                    case let .failure(error):
+                        XCTFail(error.localizedDescription)
+                    case .finished:
+                        break
+                    }
+                },
+                receiveValue: { [weak self] welcome in
+                    self?.connected.fulfill()
+                }
+            )
+    }
+
+    func testSubscribeAndReceiveEvents() throws {
+        wait(for: [connected], timeout: 0.5)
+
+        let receiveEventsExpectation = expectation(description: "3 events should have been received")
+        receiveEventsExpectation.expectedFulfillmentCount = 3
+        var receivedEvents = [Message.Event]()
+
+        session
+            .client
+            .asSubscriber?
+            .subscribe(topic: URI("com.myapp.hello")!, onUnsubscribe: { _ in })
```

Comment:
Maybe you could assert that asSubscriber actually returns something here? Otherwise this test will take 5 seconds to time out.

Reply:
or, just force unwrap? that way it will break immediately.

Reply:
Good point. Will do.